pkg/curatedpackages/regional_registry.go

```diff
@@ -76,3 +83,80 @@ func GetRegionalRegistry(defaultRegistry, region string) string {
 	}
 	return prodRegionalECRMap[region]
 }
+
+// RegistryAuthTokenProvider provides auth token for registry access.
+type RegistryAuthTokenProvider interface {
+	GetTokenByAWSConfig(ctx context.Context, awsConfig string) (string, error)
+	GetTokenByAWSKeySecret(ctx context.Context, key, secret, region string) (string, error)
+}
+
+// DefaultRegistryAuthTokenProvider provides auth token for AWS ECR registry access.
+type DefaultRegistryAuthTokenProvider struct{}
+
+// GetTokenByAWSConfig get auth token by AWS config.
+func (d *DefaultRegistryAuthTokenProvider) GetTokenByAWSConfig(ctx context.Context, awsConfig string) (string, error) {
+	file, err := os.CreateTemp("", "eksa-temp-aws-config-*")
+	if err != nil {
+		return "", err
+	}
+	if _, err := file.Write([]byte(awsConfig)); err != nil {
+		return "", err
+	}
+	defer os.Remove(file.Name())
+	if err != nil {
+		return "", err
+	}
+
+	cfg, err := config.LoadDefaultConfig(ctx,
+		config.WithSharedConfigFiles([]string{file.Name()}),
+	)
+	if err != nil {
+		return "", err
+	}
+
+	return getAuthorizationToken(cfg)
+}
+
+// GetTokenByAWSKeySecret get auth token by AWS key and secret.
+func (d *DefaultRegistryAuthTokenProvider) GetTokenByAWSKeySecret(ctx context.Context, key, secret, region string) (string, error) {
+	cfg, err := config.LoadDefaultConfig(ctx,
+		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(key, secret, "")),
+		config.WithRegion(region),
+	)
+	if err != nil {
+		return "", err
+	}
+
+	return getAuthorizationToken(cfg)
+}
+
+func getAuthorizationToken(cfg aws.Config) (string, error) {
+	ecrClient := ecr.NewFromConfig(cfg)
+	out, err := ecrClient.GetAuthorizationToken(context.Background(), &ecr.GetAuthorizationTokenInput{})
+	if err != nil {
+		return "", fmt.Errorf("ecrClient cannot get authorization token: %w", err)
+	}
+	authToken := out.AuthorizationData[0].AuthorizationToken
+	return *authToken, nil
+}
+
+// Do is a function type that takes a http request and returns a http response.
+type Do func(req *http.Request) (*http.Response, error)
+
+// TestRegistryAccessWithAWSConfig test if the AWS config has valid permission to access container registry.
+func TestRegistryAccessWithAWSConfig(ctx context.Context, awsConfig, registry string, tokenProvider RegistryAuthTokenProvider, do Do) error {
+	token, err := tokenProvider.GetTokenByAWSConfig(ctx, awsConfig)
+	if err != nil {
+		return err
+	}
```

Comment:
Actually, I missed out codecov was quite low. Could we add some tests to get up to 80% coverage?